pytorch_lightning/trainer/training_loop.py

```diff
@@ -685,71 +688,85 @@ def run_training_batch(self, batch, batch_idx, dataloader_idx):
 
         for split_idx, split_batch in enumerate(splits):
 
-            # create an iterable for optimizers and loop over them
-            for opt_idx, optimizer in optimizers:
-
-                # toggle model params + set info to logger_connector
-                self.run_train_split_start(split_idx, split_batch, opt_idx, optimizer)
-
-                result = AttributeDict()
-                if self.should_accumulate():
-                    # For gradient accumulation
-
-                    # -------------------
-                    # calculate loss (train step + train step end)
-                    # -------------------
-
-                    # automatic_optimization=True: perform dpp sync only when performing optimizer_step
-                    # automatic_optimization=False: don't block synchronization here
-                    with self.block_ddp_sync_behaviour():
-                        result = self.training_step_and_backward(
-                            split_batch, batch_idx, opt_idx, optimizer, self.trainer.hiddens
-                        )
+            if self.trainer.lightning_module.automatic_optimization:
+                for opt_idx, optimizer in self.get_active_optimizers(batch_idx):
```

Comment:
Personal, get_active_optimizers make it sound there is inactive optimizers. Might be confusing to readers.

Reply:
There are inactive optimizers. For example with two optimizers where one has frequency 1 and the other frequency 2. 
Then every other training step we only have 1 active optimizer.